binary_search_tree/binary_search_tree.py

```diff
@@ -5,20 +5,42 @@ def __init__(self, value):
     self.right = None
 
   def insert(self, value):
+    test = BinarySearchTree(value)
     if value > self.value:
       if self.right is None:
-        self.right = BinarySearchTree(value)
+        self.right = test
       else: 
-        insert(self,value)
+        self.right.insert(value)
     else: 
       if self.left is None:
-        self.left = BinarySearchTree(value)
+        self.left = test
       else: 
-        insert(self.left, value)
+        self.left.insert(value)
 
 
   def contains(self, target):
-    pass
+    if target == self.value:
+      return True
+    elif target > self.value:
+      if self.right:
```

Comment:
good work with bst bryan! hope this was a good assignment to help you understand the construct